### ISSUE TITLE
`afterCommit()` callbacks do not fire when `RefreshDatabase` handles multiple `$connectionsToTransact`

### ISSUE THREAD

**Issue by mpyw** [laravel/framework#48472]
### Laravel Version

10.24.0

### PHP Version

irrelevant

### Database Driver & Version

irrelevant

### Description

When multiple connections are specified in `$connectionsToTransact` (default value: `[null]`), any callback functions specified in `DB::afterCommit()` will not be executed.

### Steps To Reproduce

```php
<?php

namespace Tests\Feature;

use Illuminate\Foundation\Testing\RefreshDatabase;
use Illuminate\Support\Facades\DB;
use Tests\TestCase;

class ExampleTest extends TestCase
{
    protected $connectionsToTransact = [null, 'sqlite'];

    use RefreshDatabase;

    public function testExample(): void
    {
        $executed = false;

        DB::transaction(function () use (&$executed) {
            DB::afterCommit(function () use (&$executed) {
                $executed = true;
            });
        });

        $this->assertTrue($executed);
    }
}
```

**Comment by mpyw** [comment]
The timing of callback execution looks to be unclear.

- Without `RefreshDatabase`, it depends on `$transactions === 0` for each connection
- With `RefreshDatabase`, the decision may be made over multiple connections.

**Comment by mpyw** [comment]
@tonysm Does #48466 also fix #48472?

**Comment by tonysm** [comment]
It should, yeah

**Comment by tonysm** [comment]
Wait, not sure. You mentio multiple connections, I haven't tested that.

**Comment by mpyw** [comment]
Looks to be fixed by #48523 

@crynobone Thanks!

**Comment by MannikJ** [comment]
Please see #53334
